rbac/default-role-manager/role_manager.go

```diff
@@ -392,7 +430,8 @@ func (rm *RoleManagerImpl) GetRoles(name string, domains ...string) ([]string, e
 	if created {
 		defer rm.removeRole(user.name)
 	}
-	return user.getRoles(), nil
+
+	return rm.filterHelper(user.getRoles()), nil
```

Comment:
Since the `level` has been restricted in `AddLink`, will `getRoles` still exceed the level? If not, is filtering unnecessary?